### PR TITLE
checker(dm): allow unsupported charset workaround

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve?

Issue Number: ref #7333

### What is changed and how it works?

user can create table with utf8mb4 in downstream to workaround the problem that upstream table has unsupported charset

#### Tests 

 - Unit test
 - Integration test

### Release note 

```release-note
`None`.
```

### CODE REVIEW

**Review by lichunzhu — APPROVED** [review]
LGTM

**Review by GMHDBJD — APPROVED** [review]
LGTM

File: dm/tests/all_mode/run.sh [hunk 1]
```diff
@@ -575,7 +593,11 @@ function run() {
 	check_contains "all_mode"
 
 	echo "check dump files have been cleaned"
-	ls $WORK_DIR/worker1/dumped_data.$ILLEGAL_CHAR_NAME && exit 1 || echo "worker1 auto removed dump files"
+	# source1 contains unsupported charset, so dump files is uncleaned. files are
```
Comment (lichunzhu) [inline dm/tests/all_mode/run.sh:596]: where to find these code?

**Pull request merged**